svgnative/src/ports/gdiplus/GDIPlusSVGRenderer.cpp

```diff
@@ -223,14 +223,14 @@ void GDIPlusSVGRenderer::Save(const GraphicStyle& graphicStyle)
 
     if (graphicStyle.clippingPath)
     {
-        // TODO
-        SVG_ASSERT_MSG(false, "Need to implement svg clipping path!");
-    }
-
-    if (graphicStyle.opacity < 1.0f)
-    {
-        // TODO
-        SVG_ASSERT_MSG(false, "Need to implement svg opacity!");
+        std::unique_ptr<Gdiplus::GraphicsPath> clip_path(dynamic_cast<const GDIPlusSVGPath*>(graphicStyle.clippingPath->path.get())->GetGraphicsPath().Clone());
+        if (graphicStyle.clippingPath->transform)
+        {
+            const Gdiplus::Matrix* matrix = &dynamic_cast<GDIPlusSVGTransform*>(graphicStyle.clippingPath->transform.get())->GetMatrix();
+            clip_path->Transform(matrix);
+        }
+        clip_path->SetFillMode(graphicStyle.clippingPath->clipRule == WindingRule::kNonZero ? Gdiplus::FillMode::FillModeWinding : Gdiplus::FillMode::FillModeAlternate);
+        mContext->SetClip(clip_path.get());
```

Comment:
It is quite some time ago that I looked into GDI+. IIRC, clipping worked differently with GDI+ than with other rendering engines. Have you tried with SVG files where a (transformed?) group with nested graphical elements/shapes gets clipped?

```xml
<clipPath id="clip">
    <ellipse rx="25" ry="50" cx="100" cy="100"/> 
</clipPath>
<g transform="rotate(45)" clip="url(#clip)">
    <g clip="url(#clip)">
        <rect fill="green" width="200" height="200"/>
    </g>
</g>
```